certbot/tests/display/ops_test.py

```diff
@@ -37,54 +37,24 @@ def test_cancel_none(self, mock_get_utility):
         self.assertRaises(errors.Error, self._call)
         self.assertRaises(errors.Error, self._call, optional=False)
 
-    @test_util.patch_get_utility("certbot.display.ops.z_util")
-    def test_ok_safe(self, mock_get_utility):
-        mock_input = mock_get_utility().input
-        mock_input.return_value = (display_util.OK, "foo@bar.baz")
-        with mock.patch("certbot.display.ops.util.safe_email") as mock_safe_email:
-            mock_safe_email.return_value = True
-            self.assertEqual(self._call(), "foo@bar.baz")
-
-    @test_util.patch_get_utility("certbot.display.ops.z_util")
-    def test_ok_not_safe(self, mock_get_utility):
-        mock_input = mock_get_utility().input
-        mock_input.return_value = (display_util.OK, "foo@bar.baz")
-        with mock.patch("certbot.display.ops.util.safe_email") as mock_safe_email:
-            mock_safe_email.side_effect = [False, True]
-            self.assertEqual(self._call(), "foo@bar.baz")
-
     @test_util.patch_get_utility("certbot.display.ops.z_util")
     def test_invalid_flag(self, mock_get_utility):
         invalid_txt = "There seem to be problems"
         mock_input = mock_get_utility().input
         mock_input.return_value = (display_util.OK, "foo@bar.baz")
-        with mock.patch("certbot.display.ops.util.safe_email") as mock_safe_email:
-            mock_safe_email.return_value = True
-            self._call()
-            self.assertTrue(invalid_txt not in mock_input.call_args[0][0])
-            self._call(invalid=True)
-            self.assertTrue(invalid_txt in mock_input.call_args[0][0])
+        self._call()
```

Comment:
The problem with the test here is this call causes an infinite loop. You can verify this for yourself by adding print statements or using a debugger. For a recommendation for the latter, I think most people on the Certbot team like [ipdb](https://pypi.org/project/ipdb/) and it's installed in Certbot's development environment.

Reply:
If you decide to try `ipdb`, you may find the [docs for Python's standard debugger `pdb`](https://docs.python.org/3/library/pdb.html) useful since I think ipdb's docs are kind of lacking. `ipdb` can be used just like `pdb`, but it has more polish and features.